src/core_modules/capture-core-utils/rulesEngine/RulesEngine.js

```diff
@@ -0,0 +1,346 @@
+                // Replace all occurrences of the variable name(hence using regex replacement):
+                expression = expression
+                    .replace(
+                        new RegExp(`A{${variablePresent}}`, 'g'),
+                        variablesHash[variablePresent].variableValue,
+                    );
+            } else {
+                warnMessage(expression, variablePresent);
+            }
+        });
+    }
+
+    // Check if the expression contains constants
+    if (expression.indexOf('C{') !== -1) {
+        // Find every constant in the expression;
+        const variablesPresent = expression.match(/C{\w+.?\w*}/g);
+        // Replace each matched variable:
+        variablesPresent && variablesPresent.forEach((variablePresent) => {
+            // First strip away any prefix and postfix signs from the variable name:
+            variablePresent = variablePresent.replace('C{', '').replace('}', '');
+
+            if (isDefined(variablesHash[variablePresent]) && variablesHash[variablePresent].variablePrefix === 'C') {
+                // Replace all occurrences of the variable name(hence using regex replacement):
+                expression = expression
+                    .replace(
+                        new RegExp(`C{${variablePresent}}`, 'g'),
+                        variablesHash[variablePresent].variableValue,
+                    );
+            } else {
+                warnMessage(expression, variablePresent);
+            }
+        });
+    }
+
+    return expression;
+};
+
+export class RulesEngine {
+    inputConverter: IConvertInputRulesValue;
+    outputConverter: IConvertOutputRulesEffectsValue;
+    variableService: VariableService;
+    dateUtils: DateUtils;
+
+    constructor(
+        inputConverter: IConvertInputRulesValue,
+        outputConverter: IConvertOutputRulesEffectsValue,
+        momentConverter: IMomentConverter,
+    ) {
+        this.inputConverter = inputConverter;
+        this.outputConverter = outputConverter;
+        const valueProcessor = new ValueProcessor(inputConverter);
+        this.variableService = new VariableService(valueProcessor.processValue, momentConverter);
+        this.dateUtils = getDateUtils(momentConverter);
+    }
+
+    /**
+    *
+    * @param {*} programRulesContainer all program rules and program rule variables
+    * @param {*} currentEvent selected event (null if not applicable)
+    * @param {*} eventsContainer list of all relevant events, or null
+    * @param {*} dataElements all data elements (metadata)
+    * @param {*} selectedEntity selected TEI (null if not applicable)
+    * @param {*} trackedEntityAttributes all tracked entity attributes (metadata)
+    * @param {*} selectedEnrollment selected enrollment (null if not applicable)
+    * @param {*} selectedOrgUnit selected OrgUnit
+    * @param {*} optionSets all option sets
+    */
+    getProgramRuleEffects(
```

Comment:
Do you think it's better to use the object here instead of multiple params? It could be helpful when developers don't remember the order of parameters.

Reply:
So you're saying it's a pattern to pass one object containing named parameters instead of passing a comma separated list? Then one would have to remember or check the names instead of the parameter order though. But that's probably easier in this case where there are more than four parameters and the ordering is quite arbitrary.

Reply:
Yes, exactly!